### PR TITLE
[POAE7-2454] ::duckdb::QueryResult::GetConfigTimezone not found when CIDER_ENABLE_DUCKDB enabled

### PR DESCRIPTION
**Pull request by YBRua**
### What changes were proposed in this pull request?

Bug fix. In `DuckDbResultConvertor` we used duckdb's `ToArrowSchema()` method for exporting Arrow schemas. However, the versions of duckdb when `CIDER_ENABLE_VELOX` is enabled or disabled are different, and contain breaking API changes, the signature of `ToArrowSchema()` are slightly different, which leads to compile errors when `CIDER_ENABLE_VELOX=OFF`.

Since the two versions of duckdb differs in several APIs and it is troublesome to directly fix the issue, we extracted the logic of `ToArrowSchema()` from the duckdb source code and maintain a copy of this function independently (see `DuckDbArrowAdaptor.h`).

NOTE: setting `CIDER_ENABLE_VELOX=OFF` will not compile due to another error in `SubstraitExtension.cpp` (msg: error: cannot convert 'const YAML::Node' to 'bool' in return), a temporary fix to make the code at least compilable would be changing the return statement to `return false;` (`true` should also work).

### Why are the changes needed?

Bug fix, as is described above.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Re-compiled and ran UTs.

### Which label does this PR belong to?

BUG

### CODE REVIEW

File: cider/tests/utils/DuckDbQueryRunner.cpp [hunk 1]
```diff
@@ -416,15 +411,14 @@ void DuckDbResultConvertor::updateChildrenNullCounts(CiderBatch& batch) {
...
-    std::string config_timezone) {
+    std::vector<std::string>& names) {
   int col_num = chunk->ColumnCount();
   int row_num = chunk->size();
 
   // Construct ArrowSchema using methods from duckdb
   std::vector<::duckdb::LogicalType> types = chunk->GetTypes();
   auto arrow_schema = CiderBatchUtils::allocateArrowSchema();
-  ::duckdb::QueryResult::ToArrowSchema(arrow_schema, types, names, config_timezone);
+  DuckDbArrowSchemaAdaptor::duckdbResultToArrowSchema(arrow_schema, types, names);
```
Comment (kd97) [inline cider/tests/utils/DuckDbQueryRunner.cpp:422]: can move `updateChildrenNullCounts` into this method and update inside?

Comment (YBRua) [reply cider/tests/utils/DuckDbQueryRunner.cpp:422]: I'm afraid the answer is no. `null_count` is an attribute of arrow **array**, but we are dealing with arrow **schema** here, and arrow schema does not have a `null_count` attribute.

File: cider/tests/utils/DuckDbArrowAdaptor.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  static void SetArrowMapFormat(DuckDBArrowSchemaHolder& root_holder,
+                                ArrowSchema& child,
+                                const ::duckdb::LogicalType& type);
+
+  static void SetArrowFormat(DuckDBArrowSchemaHolder& root_holder,
+                             ArrowSchema& child,
+                             const ::duckdb::LogicalType& type);
+
+ public:
+  static void duckdbResultToArrowSchema(ArrowSchema* out_schema,
```
Comment (jikunshang) [inline cider/tests/utils/DuckDbArrowAdaptor.h:72]: better rename to `duckdbResultSchemaToArrowSchema`

**Review by jikunshang — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Review by jikunshang — DISMISSED** [review]

**Comment by jikunshang** [comment]
please fix code style issue.

**Review by jikunshang — DISMISSED** [review]

**Review by yma11 — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**